src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py

```diff
@@ -325,16 +325,41 @@ def lateral_statement(lateral_expression: str, child: str) -> str:
     )
 
 
-def join_table_function_statement(func: str, child: str) -> str:
+def join_table_function_statement(
+    func: str,
+    child: str,
+    left_cols: List[str],
+    right_cols: List[str],
+    use_constant_subquery_alias: bool,
+) -> str:
+    left_alias = (
+        "T_LEFT"
```

Comment:
Just curious: do we need to worry about alias conflict if there are multiple joins?

Reply:
when table joins are involved, we only allow a single table function coming in. I am not allowing aliasing the left child but I have removed it now - is not possible to cover all the cases

Reply:
We don't flatten joins so a join is always binary and subquery alias only persists at one level, so subquery aliases won't conflict.